pkg/sinks/slack.go

```diff
@@ -0,0 +1,81 @@
+package sinks
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/k8sgpt-ai/k8sgpt-operator/api/v1alpha1"
+)
+
+var _ ISink = (*SlackSink)(nil)
+
+type SlackSink struct {
+	Endpoint string
+	Client   Client
+}
+
+type SlackMessage struct {
+	Text        string       `json:"text"`
+	Attachments []Attachment `json:"attachments"`
+}
+
+type Attachment struct {
+	Type  string `json:"type"`
+	Text  string `json:"text"`
+	Color string `json:"color"`
+	Title string `json:"title"`
+}
+
+func buildSlackMessage(kind, name, details, backend string) SlackMessage {
+	return SlackMessage{
+		Text: fmt.Sprintf("`Analysis from %s of the %s %s`", backend, kind, name),
+		Attachments: []Attachment{
+			Attachment{
+				Type:  "mrkdwn",
+				Text:  details,
+				Color: "danger",
+				Title: "Report",
+			},
+		},
+	}
+}
+
+func (s *SlackSink) Configure(config v1alpha1.K8sGPT, c Client) {
+	if config.Spec.Sink == nil {
+		s.Endpoint = ""
+	}
+	s.Endpoint = config.Spec.Sink.Endpoint
+	s.Client = c
+}
+
+func (s *SlackSink) Emit(results v1alpha1.ResultSpec) error {
+	if s.Endpoint == "" {
```

Comment:
There should be some sort of error that returns a warning otherwise this will be very frustrating

Reply:
My thought so far was that an empty endpoint comes from two different places,
a) Top level Sink struct is omitted so a user don't want to configure a webhook endpoint thus it's empty
b) a user sets an empty webhook in their CR ( not sure why to set a blank CR value) , also in that case endpoint is empty so we don't emit anything,

I think a message or warning is not easy with this implementation cause we don't know which of the two cases is real, 
finally, I am not sure how a user can be frustrated if they don't set any webhook url while expecting to integrate k8sgpt, but eager to get more feedback

Reply:
I also wanted to avoid more if conditions but looks more clunky, I will update the PR